app/models/school_workforce_census.rb

```diff
@@ -115,7 +97,7 @@ class SchoolWorkforceCensus < ApplicationRecord
     ]
   }.freeze
 
-  def subjects
-    SUBJECT_ATTRIBUTES.map { |attr| send(attr) }.reject(&:blank?)
+  def subjects(trn)
```

Comment:
This would make sense if it was a class method like `subjects_by_trn`; or simply try the other suggestion above and get rid of this method completely